helix-term/src/ui/mod.rs

```diff
@@ -265,6 +266,76 @@ pub fn file_picker(root: PathBuf, config: &helix_view::editor::Config) -> FilePi
     picker
 }
 
+pub fn file_browser(root: PathBuf) -> Result<FilePicker, std::io::Error> {
+    let root = helix_stdx::path::canonicalize(root);
+    let directory_content = directory_content(&root)?;
+
+    let columns = [PickerColumn::new(
+        "path",
+        |item: &PathBuf, root: &PathBuf| {
+            let name = item.strip_prefix(root).unwrap_or(item).to_string_lossy();
+            if item.is_dir() {
+                format!("{}/", name).into()
```

Comment:
Building on the changes in #12205 we can style directories with `"ui.text.directory"`. So `file_browser` will need to be passed `&Editor` to read from the theme and then the Picker's data can be a tuple or a new struct that holds the root and the directory style. So this column definition would instead become something like:

```rust
let columns = [PickerColumn::new(
    "path",
    |(path, is_dir): &(PathBuf, bool), data: &FileBrowserData| {
        let name = path
            .strip_prefix(&data.root)
            .unwrap_or(path)
            .to_string_lossy();
        if *is_dir {
            Span::styled(format!("{}/", name), data.directory_style).into()
        } else {
            name.into()
        }
    },
)];
```

Reply:
And this styling can be applied in the Picker rendering function so that directories are also `"ui.text.directory"` in the preview